oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/hooks.js

```diff
@@ -22,45 +25,210 @@ export const useVocabularyOptions = (vocabularyType) => {
   return { options: vocabularies[vocabularyType] };
 };
 
-export const submitContextType = {
-  create: "create",
-  update: "update",
-  preview: "preview",
+// hook that enables me to highlight the related inputs when I hover over the close buttons
+// for arrayFields type of inputs
+export const useHighlightState = () => {
+  const [highlightedStates, setHighlightedStates] = React.useState([]);
+
+  const handleHover = (index) => {
+    const updatedStates = [...highlightedStates];
+    updatedStates[index] = true;
+    setHighlightedStates(updatedStates);
+  };
+
+  const handleMouseLeave = (index) => {
+    const updatedStates = [...highlightedStates];
+    updatedStates[index] = false;
+    setHighlightedStates(updatedStates);
+  };
+
+  return { highlightedStates, handleHover, handleMouseLeave };
+};
+
+export const useConfirmationModal = () => {
+  const [isModalOpen, setIsModalOopen] = React.useState(false);
+
+  const handleCloseModal = () => setIsModalOopen(false);
+  const handleOpenModal = () => setIsModalOopen(true);
+
+  return { isModalOpen, handleCloseModal, handleOpenModal };
 };
 
-export const useOnSubmit = ({
-  apiUrl,
-  context = submitContextType.create,
-  apiClient = OARepoDepositApiClient,
-  onBeforeSubmit = (values, formik) => values,
-  onSubmitSuccess = () => {},
-  onSubmitError = () => {},
-}) => {
-  const { error: submitError, mutateAsync: submitAsync } = useMutation({
-    mutationFn: async ({ apiUrl, data }) => {
-      return context === submitContextType.create
-        ? apiClient.createDraft(apiUrl, data)
-        : context === submitContextType.update
-        ? apiClient.saveDraft(apiUrl, data)
-        : new Error(`Unsupported submit context: ${context}`);
-    },
-  });
-
-  const onSubmit = (values, formik) => {
-    values = invokeCallbacks(onBeforeSubmit, values, formik);
-    submitAsync({
-      apiUrl,
-      data: values,
-    })
-      .then((result) => {
-        formik.setSubmitting(false);
-        invokeCallbacks(onSubmitSuccess, result, formik);
-      })
-      .catch((error) => {
-        formik.setSubmitting(false);
-        invokeCallbacks(onSubmitError, error);
+export const useDepositApiClient = (
+  baseApiClient,
+  serializer,
+  internalFieldsArray = [
+    "errors",
+    "validationErrors",
+    "httpErrors",
+    "successMessage",
+  ],
+  keysToRemove = ["__key"]
+) => {
+  const {
+    isSubmitting,
+    values,
+    validateForm,
+    setErrors,
+    setSubmitting,
+    setValues,
+    setFieldError,
+    setFieldValue,
+  } = useFormikContext();
```

Comment:
cant we do `const formik = useFormikCOntext` and then `const { isSubmitting, values,....} = formik` here? So that we can export the whole formik object for users of the hook?

Reply:
@mirekys You are correct, I did not think of it to do it like that. Thanks!